WordPress/src/main/java/org/wordpress/android/ui/WPWebChromeClientWithFileChooser.java

```diff
@@ -0,0 +1,119 @@
+package org.wordpress.android.ui;
+
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.view.View;
+import android.webkit.ValueCallback;
+import android.webkit.WebChromeClient;
+import android.webkit.WebView;
+import android.widget.ProgressBar;
+
+import org.wordpress.android.fluxc.utils.MediaUtils;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.AppLog.T;
+import org.wordpress.android.util.helpers.WPWebChromeClient;
+
+import static android.app.Activity.RESULT_OK;
+import static org.wordpress.android.ui.RequestCodes.WEB_CHROME_CLIENT_FILE_PICKER;
+
+public class WPWebChromeClientWithFileChooser extends WPWebChromeClient {
+    private ValueCallback<Uri[]> mFilePathCallback;
+    private final OnShowFileChooserListener mOnShowFileChooserListener;
+
+    public WPWebChromeClientWithFileChooser(Activity activity, View view, int defaultPoster,
+                                            ProgressBar progressBar,
+                                            OnShowFileChooserListener onShowFileChooserListener) {
+        super(activity, view, defaultPoster, progressBar);
+        this.mOnShowFileChooserListener = onShowFileChooserListener;
+    }
+
+    @Override public boolean onShowFileChooser(WebView webView, ValueCallback<Uri[]> filePathCallback,
+                                               FileChooserParams fileChooserParams) {
+        this.mFilePathCallback = filePathCallback;
+
+        // Check if MODE_OPEN_MULTIPLE is specified
+        Boolean canMultiselect = false;
+        if (fileChooserParams.getMode() == WebChromeClient.FileChooserParams.MODE_OPEN_MULTIPLE) {
+            canMultiselect = true;
+        }
+
+        Intent intent = fileChooserParams.createIntent();
+        intent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, canMultiselect);
+
+        String[] acceptableMimeTypes = fileChooserParams.getAcceptTypes();
+        int acceptableMimeTypesLength = acceptableMimeTypes.length;
+
+        // Sets an explicit MIME data type that states that all MIME types are supported.
+        intent.setType("*/*");
+        // Creates values for Intent.EXTRA_MIME_TYPES with the MIME types that are currently acceptable.
+        String[] resolvedMimeTypes = new String[acceptableMimeTypes.length];
```

Comment:
I think we can use `acceptableMimeTypesLength` instead of `acceptableMimeTypes.length` here.

Reply:
Makes sense 43a5c51